api/tools/src/main/java/org/pmiops/workbench/tools/ExportWorkspaceData.java

```diff
@@ -190,15 +142,79 @@ public CommandLineRunner run(
       final CustomMappingStrategy mappingStrategy = new CustomMappingStrategy();
       mappingStrategy.setType(WorkspaceExportRow.class);
 
-      StatefulBeanToCsv beanWriter =
-          new StatefulBeanToCsvBuilder(writer).withMappingStrategy(mappingStrategy).build();
+      try (FileWriter writer =
+          new FileWriter(opts.getOptionValue(exportFilenameOpt.getLongOpt()))) {
+        StatefulBeanToCsv beanWriter =
+            new StatefulBeanToCsvBuilder(writer).withMappingStrategy(mappingStrategy).build();
 
-      beanWriter.write(rows);
-
-      writer.close();
+        beanWriter.write(rows);
+      }
     };
   }
 
+  private WorkspaceExportRow toWorkspaceExportRow(DbWorkspace workspace) {
+    WorkspaceExportRow row = new WorkspaceExportRow();
+    row.setCreatorContactEmail(workspace.getCreator().getContactEmail());
+    row.setCreatorUsername(workspace.getCreator().getEmail());
+    row.setCreatorFirstSignIn(dateFormat.format(workspace.getCreator().getFirstSignInTime()));
+    row.setCreatorRegistrationState(workspace.getCreator().getDataAccessLevelEnum().toString());
+
+    row.setProjectId(workspace.getWorkspaceNamespace());
+    row.setName(workspace.getName());
+    row.setCreatedDate(dateFormat.format(workspace.getCreationTime()));
+
+    try {
+      row.setCollaborators(
+          extractAclResponse(
+                  workspacesApi.getWorkspaceAcl(
+                      workspace.getWorkspaceNamespace(), workspace.getFirecloudName()))
+              .keySet().stream()
+              .collect(Collectors.joining(",\n")));
+    } catch (ApiException e) {
+      row.setCollaborators("Error: Not Found");
+    }
+
+    Collection<DbCohort> cohorts = cohortDao.findByWorkspaceId(workspace.getWorkspaceId());
+    row.setCohortNames(
+        cohorts.stream().map(cohort -> cohort.getName()).collect(Collectors.joining(",\n")));
+    row.setCohortCount(String.valueOf(cohorts.size()));
+
+    Collection<DbConceptSet> conceptSets =
+        conceptSetDao.findByWorkspaceId(workspace.getWorkspaceId());
+    row.setConceptSetNames(
+        conceptSets.stream()
+            .map(conceptSet -> conceptSet.getName())
+            .collect(Collectors.joining(",\n")));
+    row.setConceptSetCount(String.valueOf(conceptSets.size()));
+
+    Collection<DbDataset> datasets = dataSetDao.findByWorkspaceId(workspace.getWorkspaceId());
+    row.setDatasetNames(
+        datasets.stream().map(dataSet -> dataSet.getName()).collect(Collectors.joining(",\n")));
+    row.setDatasetCount(String.valueOf(datasets.size()));
+
+    try {
+      Collection<FileDetail> notebooks =
+          notebooksService.getNotebooks(
+              workspace.getWorkspaceNamespace(), workspace.getFirecloudName());
+      row.setNotebookNames(
+          notebooks.stream()
+              .map(notebook -> notebook.getName())
+              .collect(Collectors.joining(",\n")));
```

Comment:
nit: define a constant or parameter or this joining string. Also, I'm really nervous about trying to embed sub-cell formatting this way. I'm not sure how well different tools will handle embedded newlines (and we'd want to test Windows as well I think). You just don't want to ask too much of CSV files, because lots of tools think they're compliant but I don't think there's a universal standard. But my opinion is based on experience a while back, so hopefully things are better now.